### DISCUSSION TITLE
[iOS] Questions on framerates

### DISCUSSION THREAD

**Discussion by KrisCris** [ppy/osu#15911] [Q&A] [answered]
```
Device: iPad Pro M1
OS: iOS 15.1
```

What does the number of fps of `Input`, `Audio`, `Update`, `Draw` each mean?
- I can see `Input` and `Draw` are always capped at 120fps. Does `Input` stand for the sampling rate of the touch screen? What about the external keyboard?
- I assume `Draw` is the current game's framerate,  then it is pretty low at ~60fps on the song selection page. And it randomly drops below 100 in the game, which can be fixed by launching screen recording - which is weird - maybe it's how iOS manages resources?

**Comment by peppy** [comment, marked as answer]
Nothing you state is incorrect. Except that draw FPS is iOS specific (it's not). Make sure you are running single thread and things should be performing as best they can.

Performance optimisations are an ongoing effort. You will see improvements as time passes.

**Reply by KrisCris** [reply]
Are you suggesting we should not enable multithreading on iOS devices? I did see some performance gain so I enabled it days before...

**Reply by KrisCris** [reply]
Btw, my friends told me he experienced significant latency ~30ms, I am not sure what's going on lol.

**Reply by peppy** [reply]
It is intentionally set to single thread to get better performance. Changing settings is not recommended.

Also you are self compiling and likely doing it wrong. Please wait for an official release.
